app/models/merchant.rb

```diff
@@ -1,6 +1,5 @@
 class Merchant < ApplicationRecord
-  has_many :orders
-  has_many :products
+  has_many :products, dependent: :destroy
 
```

Comment:
Do we really want to delete those or should be use nullify? 
has_many :products, dependent: :nullify

Reply:
Sure we can use nullify, I'm up for either